apps/cyberstorm-remix/app/settings/user/Settings.tsx

```diff
@@ -1,46 +1,72 @@
-import { Outlet, useLocation, useOutletContext } from "react-router";
+import {
+  Outlet,
+  useLocation,
+  useLoaderData,
+  useOutletContext,
+  type LoaderFunctionArgs,
+} from "react-router";
 import { NewLink, Tabs } from "@thunderstore/cyberstorm";
 import { PageHeader } from "~/commonComponents/PageHeader/PageHeader";
-
 import { type OutletContextShape } from "../../root";
 import "./Settings.css";
 import { NotLoggedIn } from "~/commonComponents/NotLoggedIn/NotLoggedIn";
+import {
+  getPublicEnvVariables,
+  getSessionTools,
+} from "cyberstorm/security/publicEnvVariables";
+import { DapperTs } from "@thunderstore/dapper-ts";
+
+function getSessionIdFromCookie(cookieHeader: string): string | null {
+  const match = cookieHeader
+    .split(";")
+    .map((c) => c.trim())
+    .find((c) => c.startsWith("sessionid="));
+  return match ? match.split("=")[1] : null;
+}
 
-// export async function clientLoader() {
-//   const _storage = new NamespacedStorageManager(SESSION_STORAGE_KEY);
-//   const currentUser = getSessionCurrentUser(_storage, true, undefined, () => {
-//     clearSession(_storage);
-//     throw new Response("Your session has expired, please log in again", {
-//       status: 401,
-//     });
-//     // redirect("/");
-//   });
+function tabClass(isActive: boolean) {
+  return `tabs-item${isActive ? " tabs-item--current" : ""}`;
+}
 
-//   if (
-//     !currentUser.username ||
-//     (currentUser.username && currentUser.username === "")
-//   ) {
-//     clearSession(_storage);
-//     throw new Response("Not logged in.", { status: 401 });
-//   } else {
-//     return {
-//       currentUser: currentUser as typeof currentUserSchema._type,
-//     };
-//   }
-// }
+export async function loader({ request }: LoaderFunctionArgs) {
+  const cookieHeader = request.headers.get("cookie") || "";
+  const sessionId = getSessionIdFromCookie(cookieHeader);
+  const publicEnvVariables = getPublicEnvVariables(["VITE_API_URL"]);
 
-// export function HydrateFallback() {
-//   return <div style={{ padding: "32px" }}>Loading...</div>;
-// }
+  const dapper = new DapperTs(() => {
+    return {
+      apiHost: publicEnvVariables.VITE_API_URL,
+      sessionId: sessionId || undefined,
+    };
+  });
+
+  const currentUser = await dapper.getCurrentUser();
+  return { currentUser };
+}
+
+export async function clientLoader() {
+  const tools = getSessionTools();
+  const currentUser = await tools.getSessionCurrentUser(true);
+  return { currentUser };
+}
+
+export function HydrateFallback() {
```

Comment:
In #1604 I addressed why using HydrateFallback seems like a suboptimal solution for a page with tabs. Should Suspense+Await be used here too?